src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -68,119 +74,77 @@ public boolean isRunning(){
+    // if we want to cut off whole communications, we'll do Stop
+    public boolean Disconnect(String peerId){
+        BtToRequestSocket tmpSoc = mBtToRequestSocket;
+        if(tmpSoc == null) {
+            return false;
         }
 
-        if(mBtToRequestSocket != null) {
-            print_debug("Disconnect:::Stop : mBtToRequestSocket");
-            mBtToRequestSocket.Stop();
+        String currentpeerId = tmpSoc.GetPeerId();
+        print_debug("Disconnect : " + peerId + ", current request : " + currentpeerId);
+        if (peerId.length() == 0 || peerId.equalsIgnoreCase(currentpeerId)) {
+            print_debug("Disconnect:::Stop :" + currentpeerId);
+
+            tmpSoc.Stop();
             mBtToRequestSocket = null;
+
+            return true;
+
         }
+        return false;
     }
 
-    public boolean Disconnect(String peerId){
+    //test time implementation to simulate 'peer disappearing'
+    public boolean  DisconnectIncomingConnections() {
 
-        boolean ret = false;
+        if (mServerSocketList == null) {
+            return false;
+        }
 
-// we only cut off our outgoing connections, incoming ones are cut off from the other end.
-// if we want to cut off whole communications, we'll do Stop
-        if(mBtToRequestSocket != null) {
-            String currentpeerId = mBtToRequestSocket.GetPeerId();
-            print_debug("Disconnect : " + peerId + ", current request : " + currentpeerId);
-            if(peerId.length() == 0 || peerId.equalsIgnoreCase(currentpeerId)) {
-                print_debug("Disconnect:::Stop :" + currentpeerId);
-                mBtToRequestSocket.Stop();
-                mBtToRequestSocket = null;
-                ret = true;
+        for (int i = 0; i < mServerSocketList.size(); i++) {
+            BtToServerSocket tmpToSrvSocket = mServerSocketList.get(i);
+            if (tmpToSrvSocket != null) {
+                print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmpToSrvSocket.getName());
+                tmpToSrvSocket.Stop();
+                mServerSocketList.remove(i);
             }
         }
 
-        return ret;
-    }
+        mServerSocketList.clear();
 
-    //test time implementation to simulate 'peer disappearing'
-    public boolean  DisconnectIncomingConnections(){
-
-        boolean ret = false;
-        if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
-                }
-            }
+        return true;
 
-            mServerSocketList.clear();
-        }
-        return ret;
     }
 
     public String GetBluetoothAddress(){
```

Comment:
Since this method is only used inside of BtConnectorHelper wouldn't it be easier to make this method private?

It's generally good defensive programming to make as many things private as you can. It reduces the number of places where it can cause trouble. Now, to be fair, this can be a pain when writing tests and sometimes one will either make things public or use protected for those purposes. But that doesn't apply here. So I would evaluate the methods and see what you can make private.